src/components/items.tsx

```diff
@@ -5,43 +5,81 @@ import { v4 as uuidv4 } from "uuid";
 
 export function Items() {
   const [input, setInput] = useState("");
-  const [things, setThings] = useState<TItemDocument[]>([]);
-  useEffect(() => {
+  const [things, setThings] = useState<Partial<TItemDocument>[]>([]);
+
+  const fetchItems = () =>
     itemsServices.listItems().then((res) => {
       setThings(res);
     });
+  useEffect(() => {
+    fetchItems();
   }, []);
   const addItem = () => {
     const $id = uuidv4();
     const newItem = {
       name: input,
       isSelected: false,
     };
+    setThings((prev) => [...prev, { $id, ...newItem }]);
+
```

Comment:
Nice optimistic update. However, what if the request fails? Maybe if that happens it can be removed, and show an error message somewhere